src/box/sql/show.c

```diff
@@ -0,0 +1,425 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include <ctype.h>
+#include "sqlInt.h"
+#include "schema.h"
+#include "sequence.h"
+#include "coll_id_cache.h"
+#include "tuple_constraint_def.h"
+
+enum {
+	TYPE_STR_SIZE = 16,
+};
+
+/** An objected used to accumulate a statement. */
+struct sql_desc {
+	/** Accumulate the string representation of the statement. */
+	struct StrAccum acc;
+	/** Accumulate the string representation of the error. */
+	struct StrAccum err;
+};
+
+/** Initialize the object used to accumulate a statement. */
+static void
+sql_desc_initialize(struct sql_desc *desc)
+{
+	sqlStrAccumInit(&desc->acc, NULL, 0, SQL_MAX_LENGTH);
+	sqlStrAccumInit(&desc->err, NULL, 0, SQL_MAX_LENGTH);
+}
+
+/** Append a new string to the object used to accumulate a statement. */
+static void
+sql_desc_append(struct sql_desc *desc, const char *fmt, ...)
```

Comment:
To be on the safe side I suggest to use here `CFORMAT` attribute (from `trivia/util.h`). It will validate the format string and args at compile time.

Reply:
Thanks, added.